eo-maven-plugin/src/main/java/org/eolang/maven/Footprint.java

```diff
@@ -0,0 +1,158 @@
+ *  </pre>
+ * </code>
+ * @since 1.0
+ */
+public class Footprint {
+    /**
+     * Path to target root.
+     */
+    private final Path main;
+
+    /**
+     * Version tag.
+     */
+    private final String ver;
+
+    /**
+     * Path to cache root.
+     */
+    private final Path cache;
+
+    /**
+     * Ctor.
+     * @param ver Version tag
+     * @param main Main root
+     * @param cache Cache root
+     */
+    public Footprint(final String ver, final Path main, final Path cache) {
+        this.ver = ver;
+        this.main = main;
+        this.cache = cache;
+    }
+
+    /**
+     * Get content of the program.
+     * @param program Program name
+     * @param ext File extension
+     * @return Content of a file
+     * @throws IOException In case of IO issue.
+     */
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
+    public String content(final String program, final String ext) throws IOException {
+        final Path cached = new Place(program).make(this.cache.resolve(this.safeVer()), ext);
+        final Path target = new Place(program).make(this.main, ext);
+        final IoCheckedText content;
+        if (Footprint.versioned(this.ver) && cached.toFile().exists()) {
+            content = new IoCheckedText(
+                new TextOf(cached)
+            );
+        } else {
+            content = new IoCheckedText(
+                new TextOf(target)
+            );
+        }
+        return content.asString();
+    }
+
+    /**
+     * Save content.
+     * @param program Program name
+     * @param ext File extension
+     * @param content File content
+     * @throws IOException In case of IO issues
+     */
+    public void save(final String program, final String ext, final Supplier<String> content)
+        throws IOException {
+        final Path cached = new Place(program).make(this.cache.resolve(this.safeVer()), ext);
+        final Path target = new Place(program).make(this.main, ext);
+        final String text;
+        if (Footprint.versioned(this.ver) && cached.toFile().exists()) {
+            Logger.info(
+                this,
+                "File found in cache: %s",
+                cached
+            );
+            text = this.content(program, ext);
+        } else {
+            text = content.get();
+            if (Footprint.versioned(this.ver)) {
+                new Save(
+                    text,
+                    cached
+                ).save();
+            }
+        }
+        new Save(
+            text,
+            target
+        ).save();
+    }
+
+    /**
+     * Transform version for legal path.
+     * @return Version tag
+     */
+    private String safeVer() {
+        return this.ver
+            .replaceAll("[* ]", "_");
+    }
+
+    /**
+     * Is it a true version?
+     * @param ver Version to check
+     * @return True if version has meaningful value and false otherwise
+     */
+    @SuppressWarnings("PMD.ProhibitPublicStaticMethods")
```

Comment:
@mximp this suppression is redundant, this method is not public

Reply:
Fixed